packages/app/src/systems/Transaction/hooks/useTransaction.tsx

```diff
@@ -0,0 +1,217 @@
+import type { Coin } from '@fuel-wallet/types';
```

Comment:
I think this film has room for more refactoring, creating more utility functions, and reducing the amount of `[0]` being used.

Reply:
Also, we could create a test for this hook once it has important logic, or by splitting in utilities create tests for the `utils/tx.tsx`.